lib/rules/space-infix-ops.js

```diff
@@ -8,6 +8,23 @@
 // Rule Definition
 //------------------------------------------------------------------------------
 
+const OPERATORS = [
+    "*", "/", "%", "+", "-", "<<", ">>", ">>>", "<", "<=", ">", ">=",
+    "==", "!=", "===", "!==", "&", "^", "|", "&&", "||", "=",
+    "+=", "-=", "*=", "/=", "%=", "<<=", ">>=", ">>>=", "&=", "^=", "|=",
+    "?", ":", ",", "**"
+];
+
+const operatorEnum = { enum: ["always", "never", "ignore"] };
+const schemaProperties = {
+    int32Hint: { type: "boolean" },
+    all: operatorEnum
+};
+
+OPERATORS.forEach(operator => {
+    schemaProperties[operator] = operatorEnum;
```

Comment:
What happens if the user does something like this?

```json
{
  "rules": {
    "space-infix-ops": ["error", {"all": "never", "+": "always"}]
  }
}
```

I see two possibilities:

1. Treat that as an invalid schema
1. Disallow spaces for all operators except `+`, and require spaces for `+`

It looks like the implementation currently uses option 2, which seems good to me (it allows a user to easily set a "default" with exceptions). However, may you please add some tests to ensure that this behavior works correctly?